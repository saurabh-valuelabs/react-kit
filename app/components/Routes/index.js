/* eslint-disable no-restricted-globals */
/**
 *
 * Routes
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import * as Router from 'react-router-dom';

function Routes({ routes, isLogin }) {
  return (
    <>
      <main>
        <Router.Switch>
          {routes.map(route => (
            <Router.Route
              key={`route-${route.name}`}
              path={route.path}
              exact={route.exact}
              name={route.name}
              render={props =>
                isLogin || !route.login ? (
                  <route.component
                    {...props}
                    name={route.name}
                    moduleCode={route.moduleCode}
                    moduleType={route.moduleType}
                  />
                ) : (
                  <>
                    <route.preLoginComponent {...props} name={route.name} />
                  </>
                )
              }
            />
          ))}
        </Router.Switch>
      </main>
    </>
  );
}

Routes.propTypes = {
  routes: PropTypes.array.isRequired,
  isLogin: PropTypes.bool.isRequired,
};

export default Routes;
