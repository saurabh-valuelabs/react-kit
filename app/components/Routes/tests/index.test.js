/**
 *
 * Tests for Routes
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Routes from '../index';
import passedRoutes from '../../../config/_routes';
import { DEFAULT_LOCALE } from '../../../i18n';

jest.mock('react-router-dom', () => ({
  Link: 'a',
  Route: ({ children }) =>
    typeof children === 'function' ? children({ match: false }) : 'div',
  Switch: ({ children }) =>
    typeof children === 'function' ? children({ match: false }) : 'div',
}));

describe('<Routes />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Routes routes={passedRoutes} />
      </IntlProvider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <Routes routes={passedRoutes} />
      </IntlProvider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
