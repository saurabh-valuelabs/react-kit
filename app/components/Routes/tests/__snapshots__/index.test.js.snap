// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> Should render and match the snapshot 1`] = `
<main>
  div
</main>
`;
